### ISSUE TITLE
tracks failing to display certain project and context page

### ISSUE THREAD

**Issue by thouters** [TracksApp/tracks#1941]
After adding and updating some todos, tracks failed to display the context page and project page of two todos.  The start page also gave an error, presumably since it references the malicious todos.

I identified two todos who match those, and after DELETEing them from the sqlite database, the errors disappear.  
Note: At one time I did ran out of space on the filesystem unrelated to tracks.
Note: I'm running 2.3devel (2014-05-29 10:56:31 +0200), so I probably should upgrade.

This is how I fixed it in a few minutes, I include this as a reference for whomever might have the same issue as I had).

```
root@<my server> db # sqlite3 tracks-20.sqlite3.db
sqlite> select * from projects where name like '%elektriciteit%';
(other projects ommitted, only the relevant one is shown)
134|keuring elektriciteit|24|1||active|2015-09-30 18:32:34.721249|2015-11-15 09:31:13.378510||||
sqlite> select * from contexts where name like '%winkelen%';
(other contexts ommitted, only the relevant one is shown)
44|winkelen|2|f|1|2011-03-31 18:12:23|2015-11-15 09:31:13.370402
```

Intersect both and remove table rows:

```
sqlite> select * from todos where project_id = 134 and context_id = 44;
2765|44|134|wasbenzine||2015-11-13 10:11:11.805789|||1||active||2015-11-15 07:46:11.550231|
2766|44|134|stopcontacten droogkast||2015-11-13 10:11:18.636126|||1||active||2015-11-15 09:31:13.364643|
sqlite> delete from todos where id=2765;
sqlite> delete from todos where id=2766;
sqlite> .quit
```

Note: I can't imagine why I would have set the shopping list (winkelen) item 'wasbenzine' (used to degrease surfaces) under the project 'keuring elektriciteit', it should have been under a different project.

Best regards,
Thomas

**Comment by C-Otto** [comment]
Is it possible that you added a circular dependency by accident?

**Comment by thouters** [comment]
I couldn't conclude anything from the log. (I'll append it later, don't have much time now)  However this is remarkable:
sqlite> select * from dependencies where successor_id = 2766 or predecessor_id=2766;
(no match)
sqlite> select * from dependencies where successor_id = 2765 or predecessor_id=2765;
32|2740|2765|
sqlite> select * from todos where id=2740;
2740|44||sd kaartjes||2015-10-17 08:03:25.942485|||1||pending||2015-11-15 07:46:11.815427|
I might have inadvertently performed a GUI action like drag/drop todo 2765 on top of 2740. As you can see the status is set to pending (it's displayed as such in the UI).  
Thank you,
T.

**Comment by C-Otto** [comment]
@thouters Please post the log file.

**Comment by thouters** [comment]
Here you go, I'm having the same issue again today, after drag-drop moving a todo from one context to another on the home page. I probably dropped the entry on top of another again. 
[trackslog.txt](https://github.com/TracksApp/tracks/files/159832/trackslog.txt)

**Comment by C-Otto** [comment]
I see that the error is introduced when adding a dependency, which I guess is a circular dependency. This was fixed in June 2015, and the 2.3 branch includes this in the release 2.3.0.

https://github.com/TracksApp/tracks/commit/d06b85767635cfadc091c525102002516220ab9a
https://github.com/TracksApp/tracks/issues/1754

I think your database might be corrupt at this point. It might help to identify the conflicting todos and delete the dependencies manually.

**Comment by ZeiP** [comment]
As this has apparently been fixed, closing the issue. Feel free to comment if a similar issue still exists!
